projects/components/src/utils/test/datagrid/datagrid.wo.ts

```diff
@@ -78,6 +78,20 @@ export class ClrDatagridWidgetObject extends WidgetObject<ClrDatagrid> {
         return Object.keys(this.rows[index].classes);
     }
 
+    /**
+     * Returns the native element contents within all the detail pane open.
+     */
+    getAllDetailContents(): string[] {
+        return this.findElements('clr-dg-row-detail').map(detail => detail.nativeElement);
+    }
+
+    /**
+     * Returns all the detail buttons.
+     */
+    getDetailButtons(): DebugElement[] {
```

Comment:
One of the most important rules is that you should not expose DebugElements, that will cause the calling code to access the HTML on their own, but that's the job of the WOs. This need to become private and current usages need to access primitives or custom objects